pkg/monitor/monitor.go

```diff
@@ -130,15 +142,20 @@ func (d *Daemon) initializeTask(taskName defs.MonitorTask, interval time.Duratio
 
 func (d *Daemon) singleTaskRunner(activeTask *ActiveTask) func() {
 	return func() {
+		var err error
 		d.logger.Info("Run task", slog.Any("task", activeTask.TaskName))
 		defer func() {
+			if err != nil {
+				d.logger.Error("Task failed", slog.Any("task", activeTask.TaskName), slog.Any("error", err))
+				return
+			}
 			if activeTask.Cronjob == nil {
 				return
 			}
 			nextRun, _ := activeTask.Cronjob.NextRun()
 			d.logger.Info("Finish task", slog.Any("task", activeTask.TaskName), slog.Any("next_run", nextRun))
 		}()
 
-		activeTask.Instance.Run()
+		err = activeTask.Instance.Run(context.TODO())
```

Comment:
instead of context.TODO, you can simply return func(context.Context) from singleTaskRunner and use that context.